include/zephyr/arch/riscv/arch.h

```diff
@@ -236,15 +240,23 @@ static ALWAYS_INLINE unsigned int arch_irq_lock(void)
  */
 static ALWAYS_INLINE void arch_irq_unlock(unsigned int key)
 {
+#ifdef CONFIG_RISCV_SOC_HAS_CUSTOM_IRQ_LOCK_OPS
+	z_soc_irq_unlock(key);
+#else
 	__asm__ volatile ("csrs mstatus, %0"
 			  :
 			  : "r" (key & MSTATUS_IEN)
 			  : "memory");
+#endif
 }
 
 static ALWAYS_INLINE bool arch_irq_unlocked(unsigned int key)
 {
+#ifdef CONFIG_RISCV_SOC_HAS_CUSTOM_IRQ_LOCK_OPS
+	return z_soc_irq_unlocked(key);
+#else
 	return (key & MSTATUS_IEN) != 0;
+#endif
```

Comment:
because in that case you will get a compiler warning like:

```
include/zephyr/sys/arch_interface.h:247:28: warning: 'arch_irq_lock' declared 'static' but never defined [-Wunused-function]
```

Because these functions (differently from what happens with `arch_irq_enable`) are declared as `static inline` .

Reply:
Oh, fair.